### PR TITLE
New Dev

### PR DESCRIPTION
**Pull request by mauro3**
New development

### CODE REVIEW

**Comment by pwl** [comment]
As for your comment on `function interpolate!{I<:RKIntegrator}(dstate::DenseState{I}, tout, step_out)`, I would rather not let user overloaded functions mess with `DenseState` internals.  It's better to keep them as isolated as possible, again we don't even have an idea of the interface for `DenseState`, what would you put in the docs?

**Comment by mauro3** [comment]
Ok, I'll ponder this some more.  But if you want to be able to accommodate any dense output, and you think that bits needed for dense output belong to `DenseState`, then there needs to be something.

**Comment by mauro3** [comment]
Ok, I'll revert back to the old signature of `interpolate!`. 

However, I think that this is another reason to just keep all info for dense stepping in the integrator state.  Special-casing for hermite interpolation seems not worth the bother for me. Further, consider that the two stepper we got thus far actually store the previous step anyway.  But no need to decide on this now, as it is all nicely decoupled ;-)

File: examples/iterator-implementation.jl [hunk 1]
```diff
@@ -0,0 +1,173 @@
+type EulerIntegrator <: AbstractIntegrator
```
Comment (mauro3) [inline examples/iterator-implementation.jl:1]: `immutable`

Looks good.

File: examples/iterator-implementation.jl [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+
+type EulerState{T,Y}
+    step::Step{T,Y}
+end
+
+"""
+
+Once we do that, in the `init` we should change
+`EulerState(t0,y0,dy0)` to `EulerState(Step(t0,y0,dy0))` and redefine
+`output` to `output(state::EulerState)=output(state.step)`
```
Comment (mauro3) [inline examples/iterator-implementation.jl:113]: There is this fallback too https://github.com/JuliaODE/ODE.jl/blob/8fda07ce93b7848acc8df57ec84df645a89a4e42/src/base.jl#L121

File: src/dense.jl [hunk 3]
```diff
@@ -32,18 +35,23 @@ end
...
+              #  pwl: Or DenseOutput{StiffnessSwitching{whatever}}
     opts::OP
 end
 
+# TODO: this is confusing, firs you call `solve` with `DenseOutput{I}`
+# and then you call construct it as `DenseOutput{T}`.  Also this goes
+# against the convention that we pass as much as possible as
+# options.  What if a Solver takes more than one parameter?
 function solve{I}(ivp::IVP,
                   ::Type{DenseOutput{I}};
```
Comment (pwl) [inline src/dense.jl]: when did that happen?  We were using `method=` keyword for selecting the integrator.

**Comment by pwl** [comment]
Should we drop the os X tests?  They are taking 3x as much time as Linux tests (16 minutes in total, 4 min for Linux).

**Comment by pwl** [comment]
Hopefully docs will be up once we merge this PR.

**Comment by pwl** [comment]
I added a keyword option `solver` to `solve` and `ode`, it defaults to `RKIntegratorAdaptive{:rk45}`, for those users that don't care about a method.  I also slightly refactored the Rosenbrock integrator so that trying to solve an equation with initial data other then `y0::AbstractVector` falls back to the default method for `solve` resulting in an error (it just says that it is an unsupported combination of `IVP` and `AbstractSolver`).  This is more telling errors about linear algebra or conversions.

I also changed the default `norm` to `Base.vecnorm`, `Base.norm` works only up to matrices.

I think we should merge this to see what the docs look like.

**Pull request merged**